libmamba/src/api/install.cpp

```diff
@@ -1162,5 +1181,82 @@ namespace mamba
                 }
             }
         }
+
+        void
+        get_all_pkg_info(PackageDiff::package_diff_map::value_type& pkg, solver::libsolv::Database& db)
+        {
+            auto ms = pkg.second.name + "==" + pkg.second.version + " =" + pkg.second.build_string;
+            db.for_each_package_matching(
+                specs::MatchSpec::parse(ms).value(),
+                [&](specs::PackageInfo&& pkg_info) { pkg.second = pkg_info; }
+            );
+        }
+
+        void
+        install_revision(Context& ctx, ChannelContext& channel_context, std::size_t target_revision)
+        {
+            auto exp_prefix_data = PrefixData::create(ctx.prefix_params.target_prefix, channel_context);
+            if (!exp_prefix_data)
+            {
+                throw std::runtime_error(exp_prefix_data.error().what());
+            }
+            PrefixData& prefix_data = exp_prefix_data.value();
+            auto user_requests = prefix_data.history().get_user_requests();
+
+            PackageDiff pkg_diff{};
```

Comment:
`PackageDiff` is made of members which all have default constructors, so `{}` here is superfluous.
Also, `pkg_diff` is then initialized the following line, and we prefer immediate initialization instead of construction and initilizing better, so this line should be:
```c++
const PackageDiff pkg_diff = pkg_diff.from_revision(user_requests, target_revision);
```